File: 02_suraj_bhaia/app/src/main/java/com/example/dhanajitkapali/suraj_bhaia/MainActivity.java
package com.example.dhanajitkapali.suraj_bhaia;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    Button  b1;
    TextView txt;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        b1=findViewById(R.id.button);
        txt=findViewById(R.id.textView);

        /*b1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                txt.setText("dhanjit");
            }
        }); */



    }


    public void click(View view){
        txt.setText("dhanjit");

    }
}

File: 20_intent/app/src/main/java/com/codingb/a20_intent/MainActivity.kt
package com.codingb.a20_intent

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*

const val KEY_1 = "Name"    //keys are always a string
const val KEY_2 = "Age"
const val KEY_3 = "isStudent"

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        //on pressing the button the data from this activity is sent to actovity2
        //this is an example of explicit intent
        btn.setOnClickListener {
            val i = Intent(this , MainActivity2::class.java)
            i.putExtra(KEY_1 , "Dhanajit kapali")   //these values are shared using hashmap type dataStructure
            i.putExtra(KEY_2 , 20)                  //these keys are defined above
            i.putExtra(KEY_3 , true)

            startActivity(i)
        }
    }
}